core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -307,10 +307,36 @@ object ReassignPartitionsCommand extends Logging {
   private def checkIfPartitionReassignmentSucceeded(zkUtils: ZkUtils, partitionsToBeReassigned: Map[TopicAndPartition, Seq[Int]])
   :Map[TopicAndPartition, ReassignmentStatus] = {
     val partitionsBeingReassigned = zkUtils.getPartitionsBeingReassigned().mapValues(_.newReplicas)
-    partitionsToBeReassigned.keys.map { topicAndPartition =>
-      (topicAndPartition, checkIfPartitionReassignmentSucceeded(zkUtils, topicAndPartition, partitionsToBeReassigned,
-        partitionsBeingReassigned))
-    }.toMap
+    partitionsToBeReassigned.keys.filter { topicAndPartition =>
+      !partitionsBeingReassigned.contains(topicAndPartition)
+    }.groupBy { topicAndPartition =>
+      topicAndPartition.topic
+    }.flatMap { topicAndPartition =>
+      checkIfPartitionReassignmentSucceeded(zkUtils, topicAndPartition._1, topicAndPartition._2, partitionsToBeReassigned)
+    } ++ partitionsToBeReassigned.keys.filter { topicAndPartition =>
```

Comment:
could you please consider using partition method from scala collection?
it would like as
```
val (reassigned, beingReassigned) = partitionsToBeReassigned.keys.partition { topicAndPartition =>
   partitionsBeingReassigned.contains(topicAndPartition)
}
```
So you don't need to repeat your logic for filtering twice.